proxy/http2/Http2ConnectionState.cc

```diff
@@ -1677,6 +1677,8 @@ Http2ConnectionState::send_a_data_frame(Http2Stream *stream, size_t &payload_len
     payload_length = 0;
   }
 
+  stream->update_sent_count(payload_length);
```

Comment:
The condition of setting END_STREAM flag is `stream->is_write_vio_done() && !resp_reader->is_read_avail_more_than(0)`  (line 1691).

This change affects the first condition but has no effect on the second condition.

Reply:
Correct.  Assuming there is no more data to read, then promoting the update_sent_count() will cause both cases on the test on line 1691 to be true.

Reply:
I'm +1 to set the EOS flag when it's possible. But I can't imagine there're no more data to read but `payload_length` is not 0 because of this logic.

```
1663   if (resp_reader->is_read_avail_more_than(0)) {
...
1676   } else {
1677     payload_length = 0;
1678   }
```

Reply:
Hmm, you are right the extra check on  !resp_reader->is_read_avail_more_than(0) before setting the EOS byte means nothing will really change.

Seems like we shouldn't need that check either.  Let me work with that a bit more and close or update as necessary.